src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Extensions/ConditionsLinqExtentions.cs

```diff
@@ -0,0 +1,270 @@
+                        Expression.GreaterThan(propertyExpression, valueExpression));
+                case Condition.GreaterThanOrEqualTo:
+                    return Expression.AndAlso(Expression.Not(isNullExpression),
+                        Expression.GreaterThanOrEqual(propertyExpression, valueExpression));
+                case Condition.LessThan:
+                    return Expression.AndAlso(Expression.Not(isNullExpression),
+                        Expression.LessThan(propertyExpression, valueExpression));
+                case Condition.LessThanOrEqualTo:
+                    return Expression.AndAlso(Expression.Not(isNullExpression),
+                        Expression.LessThanOrEqual(propertyExpression, valueExpression));
+                case Condition.IsNull:
+                    return isNullExpression;
+                case Condition.NotIsNull:
+                    return Expression.Not(isNullExpression);
+                default:
+                    throw new NotSupportedException(string.Format(
+                        LanguageHelper.GetString("FilterBuilder_Exception_Message_ConditionIsNotSupported_Pattern"),
+                        condition));
+            }
+        }
+
+        private static Expression BuildExpression(this BooleanExpression expression,
+            ParameterExpression parameterExpression, string propertyName)
+        {
+            switch (expression.SelectedCondition)
+            {
+                case Condition.EqualTo:
+                    return Expression.AndAlso(Expression.Not(BuildIsNullExpression(parameterExpression, propertyName)),
+                        Expression.Equal(BuildPropertyExpression(parameterExpression, propertyName),
+                            Expression.Constant(expression.Value)));
+                default:
+                    throw new NotSupportedException(string.Format(
+                        LanguageHelper.GetString("FilterBuilder_Exception_Message_ConditionIsNotSupported_Pattern"),
+                        expression.SelectedCondition));
+            }
+        }
+
+        private static Expression BuildExpression(this StringExpression expression,
+            ParameterExpression parameterExpression, string propertyName)
+        {
+            return BuildStringExpression(expression.SelectedCondition, Expression.Property(parameterExpression, propertyName), Expression.Constant(expression.Value));
+        }
+
+        /// Review: Try to combine expression builders as much as possible.
+        private static Expression BuildStringExpression(Condition condition, Expression propertyExpression,
+            Expression valueExpression)
+        {
+            switch (condition)
+            {
+                case Condition.Contains:
+                    return Expression.AndAlso(Expression.Not(Expression.Call(typeof(string).GetMethod("IsNullOrEmpty", new[] { typeof(string) }), propertyExpression)),
```

Comment:
@alexfdezsauco we need to use the 5.2 ArrayShim here in the future. Please create a ticket so we don't forget.